apps/web/core/hooks/context/app-rail-context.tsx

```diff
@@ -0,0 +1,47 @@
+"use client";
+
+import React, { createContext, ReactNode } from "react";
+import { observer } from "mobx-react";
+import { useParams } from "next/navigation";
+// hooks
+import useLocalStorage from "@/hooks/use-local-storage";
+
+export interface AppRailContextType {
```

Comment:
Any specific reason why we are using React.Context here instead of Mobx store?

Reply:
We already have some logic related to SideRailMap in a mobx store. Why now use the same?